### PR TITLE
[DataGrid] Add `initialOpen` prop to `GridEditSingleSelectCell` to allow overriding default behavior on cell edit mode

### LINKED ISSUE #5502: [data grid] Using singleSelect with cell edit mode
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When combining singleselect with cell edit mode the selects are open when a record is added. This is okay with one but if you have 2 they both pop open after clicking add

image

### Expected behavior 🤔

singleselects stay closed. Row edit mode works correctly

### Steps to reproduce 🕹

https://codesandbox.io/s/fullfeaturedcrudgrid-demo-mui-x-forked-dv48zi?file=/demo.tsx

1. Click Add Record

### Context 🔦

_No response_

### Your environment 🌎

 System:
    OS: macOS 12.3.1
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.5.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Not Found
    Firefox: 94.0.1
    Safari: 15.4
  npmPackages:
    @emotion/react: 11.9.0 => 11.9.0 
    @emotion/styled: 11.8.1 => 11.8.1 
    @mui/base:  5.0.0-alpha.76 
    @mui/icons-material: 5.6.1 => 5.6.1 
    @mui/lab: 5.0.0-alpha.77 => 5.0.0-alpha.77 
    @mui/material: 5.6.1 => 5.6.1 
    @mui/private-theming:  5.1.1 
    @mui/styled-engine:  5.6.1 
    @mui/styles: 5.6.1 => 5.6.1 
    @mui/system:  5.6.1 
    @mui/types:  7.1.3 
    @mui/utils: 5.6.1 => 5.6.1 
    @mui/x-data-grid: 5.12.3 => 5.12.3 
    @mui/x-data-grid-generator: 5.12.3 => 5.12.3 
    @mui/x-data-grid-premium:  5.12.3 
    @mui/x-data-grid-pro: 5.12.3 => 5.12.3 
    @mui/x-date-pickers:  5.0.0-alpha.0 
    @mui/x-license-pro:  5.12.3 
    @types/react: ^17.0.19 => 17.0.35 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    typescript: 4.4.4 => 4.4.4 

### Order ID 💳 (optional)

39468

### PR DESCRIPTION
**Pull request by shapaaa**
This PR fixes #5502

### CODE REVIEW

File: packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx [hunk 1]
```diff
@@ -60,14 +62,14 @@ function GridEditSingleSelectCell(props: GridEditSingleSelectCellProps) {
     isProcessingProps,
     error,
     onValueChange,
+    initialOpen = rootProps.editMode === 'cell',
```
Comment (m4theushw) [inline packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx:65]: We have an enum for edit modes.

```suggestion
    initialOpen = rootProps.editMode === GridEditModes.Cell,
```

**Review by m4theushw — COMMENTED** [review]
Thanks for the contribution. You only need to run `yarn proptypes` and commit the changes for the CI to be green.

File: packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx (hunk 1 above)
Comment (shapaaa) [reply packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx]: Made suggested change

File: packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx [hunk 2]
```diff
@@ -38,9 +38,11 @@ export interface GridEditSingleSelectCellProps
    * @returns {Promise<void> | void} A promise to be awaited before calling `apiRef.current.setEditCellValue`
    */
   onValueChange?: (event: SelectChangeEvent<any>, newValue: any) => Promise<void> | void;
+  initialOpen?: boolean;
```
Comment (m4theushw) [inline packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx:41]: Could you fill the description? Once we have that it's good to be merged.

Comment (shapaaa) [reply packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx:41]: circle_ci test_static was failing when I added description
image

Comment (m4theushw) [reply packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx:41]: You need to run `yarn proptypes` and commit the result too. The proptypes are generated by reading the component's prop interface.

Comment (shapaaa) [reply packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx:41]: Yeah I already ran `yarn proptyes` and then added comment over generated proptype for description.

Comment (shapaaa) [reply packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx:41]: In the first commit I did add description above proptype as a comment that's when test_static was failing then I removed it

Comment (m4theushw) [reply packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx:44]: I fixed the description.

**Review by m4theushw — APPROVED** [review]

**Pull request merged**